### PR TITLE
Added support for getting configuration from a URL

### LINKED ISSUE #480: Add ability to load a config file from a URL and use that for installation
This would compliment the feature to use a configuration file

### PR DESCRIPTION
**Pull request by dylanmtaylor**
This will close #480. Adds support for retrieving a configuration json file from a URL as well as a filename.

### CODE REVIEW

**Comment by dylanmtaylor** [comment]
Work in progress, but to test, you can use this:
```
python -m archinstall --config https://files.dylanmtaylor.com/config-sample.json --silent
```

**Comment by l4zy0n3** [comment]
Can we try to directly parse the response as json using the requests library, that way we won't have to worry about creating temp files

File: archinstall/__init__.py [hunk 1]
```diff
@@ -41,7 +45,15 @@ def initialize_arguments():
 	args = parser.parse_args()
 	if args.config is not None:
 		try:
-			config = json.load(args.config)
+			# First, let's check if this is a URL scheme instead of a filename
+			parsed_url = urllib.parse.urlparse(args.config)
+
+			if not parsed_url.scheme:  # The Profile was not a direct match on a remote URL, it must be a local file.
+				config = json.load(args.config)
```
Comment (l4zy0n3) [inline archinstall/__init__.py:52]: I think we might need to create the file object here if we're not using FileType, as FileType automatically parses the argument as a file and creates a file object for us

**Comment by dylanmtaylor** [comment]
> Can we try to directly parse the response as json using the requests library, that way we won't have to worry about creating temp files

Fixed in latest commit.

File: archinstall/__init__.py (hunk 1 above)
Comment (dylanmtaylor) [reply archinstall/__init__.py]: Yes, I think it's correct now.

**Comment by dylanmtaylor** [comment]
Seems to be working now, tested with both file and URL.

**Pull request merged**